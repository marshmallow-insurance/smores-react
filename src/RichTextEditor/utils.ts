import { $createParagraphNode, LexicalNode, RootNode } from 'lexical'

export const appendNodes = (root: RootNode, nodes: LexicalNode[]) => {
  nodes
    .filter((node) => node.__type !== 'linebreak')
    .map((node) => {
      if (node.__type === 'text') {
        const paragraphNode = $createParagraphNode()
        paragraphNode.append(node)
        return paragraphNode
      }
      return node
    })
    .forEach((node) => root.append(node))
}
