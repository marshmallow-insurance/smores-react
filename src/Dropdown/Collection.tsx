import React, { FC, ReactNode } from 'react'
import styled from 'styled-components'

import { Text } from '../Text'
import { Box } from '../Box'
import { Dropdown, DropdownProps } from './Dropdown'
import { noop } from '../utils/noop'

const days = [
  {
    label: 'Monday',
    value: 'MONDAY',
  },
  {
    label: 'Tuesday',
    value: 'TUESDAY',
  },
  {
    label: 'Wednesday',
    value: 'WEDNESDAY',
  },
  {
    label: 'Thursday',
    value: 'THURSDAY',
  },
  {
    label: 'Friday',
    value: 'FRIDAY',
  },
  {
    label: 'Saturday',
    value: 'SATURDAY',
  },
  {
    label: 'Sunday',
    value: 'SUNDAY',
  },
]

const variationList: DropdownProps[] = [
  {
    id: 'days',
    list: days,
    placeholder: 'Placeholder',
    errorMsg: 'This field is required',
    label: 'Days of the week',
    onSelect: noop,
  },
]

export const CollectionPage: FC = () => {
  return (
    <Box flex direction="column">
      {variationList.map(
        ({ id, list, placeholder, errorMsg, label, onSelect }) => {
          return (
            <Box key={id} flex direction="column" mb="32px">
              <Title tag="span" typo="header-medium">
                Dropdown variations
              </Title>
              <Row label="Generic">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  list={list}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Fallback">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  list={list}
                  fallbackStyle={true}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Disabled">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  list={list}
                  disabled={true}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Disabled Fallback">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  list={list}
                  disabled={true}
                  onSelect={onSelect}
                  fallbackStyle
                />
              </Row>
              <Row label="Placeholder">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  list={list}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="No placeholder">
                <Dropdown
                  id={id}
                  label={label}
                  list={list}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="show default option">
                <Dropdown
                  id={id}
                  label={label}
                  list={list}
                  placeholder={placeholder}
                  showDefaultOption
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="show default option with custom option label">
                <Dropdown
                  id={id}
                  label={label}
                  list={list}
                  placeholder={placeholder}
                  showDefaultOption
                  customDefaultOption="Select a day"
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>

              <Row label="Leading Icon">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  frontIcon="iphone"
                  list={list}
                  fallbackStyle={true}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Required">
                <Dropdown
                  id={id}
                  label={label}
                  required={true}
                  placeholder={placeholder}
                  list={list}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Assistive text">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  assistiveText="Some assistive text"
                  list={list}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Completed">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  list={list}
                  error={false}
                  completed={true}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="As title">
                <Dropdown
                  id={id}
                  label={label}
                  renderAsTitle={true}
                  placeholder={placeholder}
                  list={list}
                  error={false}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Error">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  assistiveText="Some assistive text"
                  list={list}
                  error={true}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
              <Row label="Fallback Error">
                <Dropdown
                  id={id}
                  label={label}
                  placeholder={placeholder}
                  assistiveText="Some assistive text"
                  list={list}
                  fallbackStyle={true}
                  error={true}
                  errorMsg={errorMsg}
                  onSelect={onSelect}
                />
              </Row>
            </Box>
          )
        },
      )}
    </Box>
  )
}

const Row: FC<{ label: string; children?: ReactNode }> = ({
  label,
  children,
}) => {
  return (
    <RowWrapper mt="32px">
      <Box width="140px">
        <Text tag="span" typo="label-large">
          {label}
        </Text>
      </Box>
      {children}
    </RowWrapper>
  )
}

const RowWrapper = styled(Box)`
  display: flex;
  flex-direction: row;
  gap: 16px;
`

const LABEL_WIDTH = '140px'

const Title = styled(Text)`
  margin-bottom: 12px;
  font-size: 18px;
  line-height: 20px;
  margin-left: ${LABEL_WIDTH};
`
