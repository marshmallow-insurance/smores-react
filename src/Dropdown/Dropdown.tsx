import React, {
  FocusEvent,
  FormEvent,
  ForwardedRef,
  forwardRef,
  useMemo,
} from 'react'
import styled, { css } from 'styled-components'

import { Box } from '../Box'
import { Icon } from '../Icon'
import { Icons } from '../Icon/iconsList'

import { Field } from '../fields/Field'
import { CommonFieldProps } from '../fields/commonFieldTypes'
import { StyledFrontIcon } from '../fields/components/CommonInput'
import { useUniqueId } from '../utils/id'
import { useControllableState } from '../utils/useControlledState'

export type DropdownItem = {
  optionGroupLabel?: string
  label: string
  value: string
}

export interface Props extends CommonFieldProps {
  placeholder?: string
  showDefaultOption?: boolean
  customDefaultOption?: string
  name?: string
  value?: string | null
  defaultValue?: string
  disabled?: boolean
  list: DropdownItem[]
  frontIcon?: Icons
  fallbackStyle?: boolean
  onSelect: (element: string) => void
  onBlur?: (e: FocusEvent<HTMLSelectElement>) => void
}

type TruncateProps =
  | {
      onSelect: (e: string) => void
      onInputChange?: (e: FormEvent<HTMLSelectElement>) => void
    }
  | {
      onSelect?: (e: string) => void
      onInputChange: (e: FormEvent<HTMLSelectElement>) => void
    }

export type DropdownProps = Props & TruncateProps

export const Dropdown = forwardRef(function Dropdown(
  {
    id: idProp,
    placeholder,
    showDefaultOption = false,
    customDefaultOption,
    name,
    value: valueProp,
    defaultValue,
    disabled = false,
    list,
    onSelect,
    error = false,
    onInputChange,
    onBlur,
    frontIcon,
    fallbackStyle,
    ...fieldProps
  }: DropdownProps,
  ref: ForwardedRef<HTMLSelectElement>,
) {
  const [value, setValue] = useControllableState({
    initialState: defaultValue,
    stateProp: valueProp,
  })
  const id = useUniqueId(idProp)
  const hasOptGroups = list.findIndex((item) => !!item.optionGroupLabel) !== -1

  const dropdownItemsGroups = useMemo(() => {
    const itemsPerGroupLabel = new Map<string, DropdownItem[]>()

    list.forEach((item) => {
      const key = item.optionGroupLabel ?? ''
      const group = itemsPerGroupLabel.get(key) ?? []

      group.push(item)
      itemsPerGroupLabel.set(key, group)
    })

    return Array.from(itemsPerGroupLabel.values())
  }, [list])

  const defaultOptionLabel = () => {
    if (!showDefaultOption) {
      return placeholder
    }
    return customDefaultOption ?? 'Select an option'
  }

  return (
    <Field {...fieldProps} htmlFor={id} error={error}>
      <Box flex alignItems="center" style={{ position: 'relative' }}>
        {frontIcon && (
          <StyledFrontIcon
            $disabled={disabled}
            render={frontIcon}
            color="sesame"
          />
        )}
        <StyledSelect
          id={id}
          disabled={disabled || list.length < 1}
          onChange={(event) => {
            const value = event.currentTarget.value

            onSelect?.(value)
            onInputChange?.(event)
            setValue(value)
          }}
          $error={error}
          ref={ref}
          onBlur={onBlur}
          name={name}
          $frontIcon={frontIcon}
          $fallbackStyle={fallbackStyle}
          value={value ?? ''}
        >
          {hasOptGroups ? (
            <optgroup label={defaultOptionLabel()}>
              <option value="" hidden={!showDefaultOption} disabled>
                {defaultOptionLabel()}
              </option>
            </optgroup>
          ) : (
            <option value="" hidden={!showDefaultOption} disabled>
              {defaultOptionLabel()}
            </option>
          )}

          {dropdownItemsGroups.map((groupItems, i) =>
            hasOptGroups ? (
              <optgroup
                key={i}
                label={groupItems[0].optionGroupLabel ?? 'Other'}
              >
                {groupItems.map((el, j) => (
                  <option key={`${i}-${j}`} value={el.value}>
                    {el.label}
                  </option>
                ))}
              </optgroup>
            ) : (
              groupItems.map((el, j) => (
                <option key={j} value={el.value}>
                  {el.label}
                </option>
              ))
            ),
          )}
        </StyledSelect>
        <Caret>
          <Icon render="caret" color="marzipan" size={24} />
        </Caret>
      </Box>
    </Field>
  )
})

const resetSelect = css`
  border: none;
  border-radius: 0;
  font-size: 16px;
  appearance: none; /* remove default arrow */
  outline: none;
`

interface SSelect {
  $error: boolean
  $frontIcon?: string
  $fallbackStyle?: boolean
  value?: string | null
}

const StyledSelect = styled.select<SSelect>`
  ${resetSelect}
  width: 100%;
  height: 32px;
  color: ${({ value, theme }) =>
    value === '' ? theme.color.text.subtle : theme.color.text.base};
  cursor: pointer;
  background-color: ${({ $fallbackStyle, theme }) =>
    $fallbackStyle
      ? theme.color.surface.base[300]
      : theme.color.background['000']};

  border-radius: 12px;
  padding: 18px 14px;
  border: 2px solid
    ${({ $error, theme }) =>
      $error ? theme.color.feedback.negative[200] : theme.color.border.subtle};
  height: auto;

  ${({ $frontIcon }) =>
    $frontIcon &&
    $frontIcon != '' &&
    `
      padding-left: 42px;
    `}

  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }

  ${({ $error }) =>
    !$error &&
    css`
      :not(:disabled) {
        &:hover,
        &:focus,
        &:focus-visible,
        &:checked {
          border-color: ${({ theme }) => theme.color.icon.nonEssential};
      }
    `}
`

const Caret = styled.div`
  position: absolute;
  z-index: 1;
  right: 15px;
  pointer-events: none;
  color: ${({ theme }) => theme.color.icon.nonEssential};
`
