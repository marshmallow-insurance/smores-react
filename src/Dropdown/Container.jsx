import React, {useState} from 'react';
import {Dropdown} from './Dropdown';

const days = [
  {
    label: 'Monday',
    value: 'Monday',
  },
  {
    label: 'Tuesday',
    value: 'Tuesday',
  },
  {
    label: 'Wednesday',
    value: 'Wednesday',
  },
  {
    label: 'Thursday',
    value: 'Thursday',
  },
  {
    label: 'Friday',
    value: 'Friday',
  },
  {
    label: 'Saturday',
    value: 'Saturday',
  },
  {
    label: 'Sunday',
    value: 'Sunday',
  },
];

export const Container = () => {
  const [day, setDay] = useState('');

  return (
    <div style={{width: '260px'}}>
      <Dropdown
        id="days"
        label="Day"
        placeholder="Select Day"
        value={day}
        list={days}
        onSelect={item => setDay(item.label)}
      />
    </div>
  );
};
