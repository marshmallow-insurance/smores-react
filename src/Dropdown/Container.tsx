import React, { useState } from 'react'
import { Box } from '../Box'
import { Dropdown } from './Dropdown'

const generateYears = () => {
  const currYear = new Date().getFullYear()

  const yearsArray = Array.from({ length: 30 }).map((_, i) => ({
    value: String(currYear - i),
    label: String(currYear - i),
  }))

  return yearsArray
}

const months = [
  {
    value: 'January',
    label: 'January',
  },
  {
    value: 'February',
    label: 'February',
  },
  {
    value: 'March',
    label: 'March',
  },
  {
    value: 'April',
    label: 'April',
  },
  {
    value: 'May',
    label: 'May',
  },
  {
    value: 'June',
    label: 'June',
  },
  {
    value: 'July',
    label: 'July',
  },
  {
    value: 'August',
    label: 'August',
  },
  {
    value: 'September',
    label: 'September',
  },
  {
    value: 'October',
    label: 'October',
  },
  {
    value: 'November',
    label: 'November',
  },
  {
    value: 'December',
    label: 'December',
  },
]

const options = [
  {
    value: 'Yes',
    label: 'Yes',
  },
]

const time = [
  {
    value: 'One',
    label: 'One',
  },
  {
    value: 'Two',
    label: 'Two',
  },
  {
    value: 'Three',
    label: 'Three',
  },
]

export const Container = () => {
  const [year, setYear] = useState('')
  const [month, setMonth] = useState('July')
  const [option, setOption] = useState('')

  return (
    <>
      <Box flex>
        <Dropdown
          id="years"
          label="Years"
          placeholder="Select year"
          list={generateYears()}
          onSelect={(year) => setYear(year)}
        />

        <Box mr="16px" />

        <Dropdown
          id="months"
          label="Months"
          placeholder="Select month"
          defaultValue={month}
          list={months}
          onSelect={(month) => setMonth(month)}
        />

        <Box mr="16px" />

        <Dropdown
          id="options"
          label="Options"
          placeholder="Select option"
          list={year ? time : options}
          onSelect={(option) => setOption(option)}
        />
      </Box>

      <Box>
        <p>{year}</p>
        <p>{month}</p>
        <p>{option}</p>
      </Box>
    </>
  )
}
