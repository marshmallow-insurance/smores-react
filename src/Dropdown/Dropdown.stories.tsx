import React from 'react'
import { Dropdown, DropdownProps } from './Dropdown'
import { CollectionPage } from './Collection'

const days = [
  {
    label: 'Monday',
    value: 'MONDAY',
  },
  {
    label: 'Tuesday',
    value: 'TUESDAY',
  },
  {
    label: 'Wednesday',
    value: 'WEDNESDAY',
  },
  {
    label: 'Thursday',
    value: 'THURSDAY',
  },
  {
    label: 'Friday',
    value: 'FRIDAY',
  },
  {
    label: 'Saturday',
    value: 'SATURDAY',
  },
  {
    label: 'Sunday',
    value: 'SUNDAY',
  },
]

export default {
  title: 'Dropdown',
  component: Dropdown,
  argTypes: { onSelect: { action: 'selected' } },
}

const Template = (props: DropdownProps) => <Dropdown {...props} />

export const Default = Template.bind({})

Default.args = {
  id: 'days',
  list: days,
  placeholder: 'Placeholder',
}

export const DefaultFallback = Template.bind({})

DefaultFallback.args = {
  id: 'days',
  list: days,
  fallback: true,
  placeholder: 'Placeholder',
}

export const Collection = CollectionPage.bind({})
