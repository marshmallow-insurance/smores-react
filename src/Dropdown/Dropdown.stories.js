import React from 'react'
import { Dropdown } from './Dropdown'
import { Container } from './Container'
import { theme } from '../theme'

const days = [
  {
    label: 'Monday',
    value: 'MONDAY',
  },
  {
    label: 'Tuesday',
    value: 'TUESDAY',
  },
  {
    label: 'Wednesday',
    value: 'WEDNESDAY',
  },
  {
    label: 'Thursday',
    value: 'THURSDAY',
  },
  {
    label: 'Friday',
    value: 'FRIDAY',
  },
  {
    label: 'Saturday',
    value: 'SATURDAY',
  },
  {
    label: 'Sunday',
    value: 'SUNDAY',
  },
]

const daysWithGroups = [
  {
    label: 'Monday',
    value: 'MONDAY',
    optionGroupLabel: 'Weekday',
  },
  {
    label: 'Tuesday',
    value: 'TUESDAY',
    optionGroupLabel: 'Weekday',
  },
  {
    label: 'Wednesday',
    value: 'WEDNESDAY',
    optionGroupLabel: 'Weekday',
  },
  {
    label: 'Thursday',
    value: 'THURSDAY',
    optionGroupLabel: 'Weekday',
  },
  {
    label: 'Friday',
    value: 'FRIDAY',
    optionGroupLabel: 'Weekday',
  },
  {
    label: 'Saturday',
    value: 'SATURDAY',
    optionGroupLabel: 'Weekend',
  },
  {
    label: 'Sunday',
    value: 'SUNDAY',
    optionGroupLabel: 'Weekend',
  },
]

export default {
  title: 'Dropdown',
  component: Dropdown,
  argTypes: { onSelect: { action: 'selected' } },
}

const Template = (args) => <Dropdown {...args} />

export const Default = Template.bind({})

Default.args = {
  id: 'days',
  list: days,
  placeholder: 'Select Day',
}

export const DefaultWithGroups = Template.bind({})

DefaultWithGroups.args = {
  id: 'days',
  list: daysWithGroups,
  placeholder: 'Select Day',
}

export const WithLabel = Template.bind({})

WithLabel.args = {
  id: 'days',
  list: days,
  placeholder: 'Select Day',
  label: 'Day',
  outlined: false,
}

export const WithLabelWithGroups = Template.bind({})

WithLabelWithGroups.args = {
  id: 'days',
  list: daysWithGroups,
  placeholder: 'Select Day',
  label: 'Day',
  outlined: false,
}

export const Disabled = Template.bind({})

Disabled.args = {
  id: 'days',
  list: days,
  placeholder: 'Select Day',
  disabled: true,
}

export const DisabledWithGroups = Template.bind({})

DisabledWithGroups.args = {
  id: 'days',
  list: daysWithGroups,
  placeholder: 'Select Day',
  disabled: true,
}

export const SingleListItem = Template.bind({})

SingleListItem.args = {
  id: 'days',
  list: days.slice(0, 1),
  placeholder: 'Select Day',
}

export const SingleListItemWithGroups = Template.bind({})

SingleListItemWithGroups.args = {
  id: 'days',
  list: daysWithGroups.slice(0, 1),
  placeholder: 'Select Day',
}

export const EmptyList = Template.bind({})

EmptyList.args = {
  id: 'days',
  list: [],
  placeholder: 'Select Day',
}

export const WithOutline = Template.bind({})

WithOutline.args = {
  id: 'days',
  list: days,
  placeholder: 'Select Day',
  outlined: true,
}

export const WithHeaderLabel = Template.bind({})

WithHeaderLabel.args = {
  id: 'days',
  list: days,
  placeholder: 'Select Day',
  outlined: true,
  label: 'What day were you born?',
  labelColor: theme.colors.secondary,
  labelTypo: 'header-medium',
  labelTag: 'h3',
  outlinedLabelMargin: '16px',
}

export const WithOutlineWithGroups = Template.bind({})

WithOutlineWithGroups.args = {
  id: 'days',
  list: daysWithGroups,
  placeholder: 'Select Day',
  outlined: true,
}

export const OutlineWithError = Template.bind({})

OutlineWithError.args = {
  id: 'days',
  list: days,
  placeholder: 'Select',
  outlined: true,
  error: true,
  errorMsg: 'This field is required',
  label: 'Select day',
}

export const OutlineWithErrorWithGroups = Template.bind({})

OutlineWithErrorWithGroups.args = {
  id: 'days',
  list: daysWithGroups,
  placeholder: 'Select',
  outlined: true,
  error: true,
  errorMsg: 'This field is required',
  label: 'Select day',
}

export const WithError = Template.bind({})

WithError.args = {
  id: 'days',
  list: days,
  placeholder: 'Select',
  outlined: false,
  error: true,
  errorMsg: 'This field is required',
  label: 'Select day',
}

export const WithErrorWithGroups = Template.bind({})

WithErrorWithGroups.args = {
  id: 'days',
  list: daysWithGroups,
  placeholder: 'Select',
  outlined: false,
  error: true,
  errorMsg: 'This field is required',
  label: 'Select day',
}

const DateSelectorExampleTemplate = (args) => <Container {...args} />

export const DateSelectorExample = DateSelectorExampleTemplate.bind({})

DateSelectorExample.args = {}
