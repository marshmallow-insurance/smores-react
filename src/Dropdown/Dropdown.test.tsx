import React from 'react'
import { expect, it } from 'vitest'
import { render } from '@testing-library/react'
import { Dropdown } from './Dropdown'
import { noop } from '../utils/noop'

const items = [
  { label: 'Option 1', value: 'option1' },
  { label: 'Option 2', value: 'option2' },
]

describe('Dropdown', () => {
  it('renders correctly with a list of items', () => {
    const { container } = render(
      <Dropdown list={items} onSelect={noop} onInputChange={noop} />,
    )
    expect(container).toMatchSnapshot()
  })
  it('renders correctly when completed', () => {
    const { container } = render(
      <Dropdown list={items} onSelect={noop} onInputChange={noop} completed />,
    )
    expect(container).toMatchSnapshot()
  })
  it('renders correctly with an error message', () => {
    const { container } = render(
      <Dropdown
        list={items}
        onSelect={noop}
        onInputChange={noop}
        error
        errorMsg="This is an error"
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
