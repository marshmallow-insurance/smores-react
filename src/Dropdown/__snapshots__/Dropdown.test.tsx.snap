// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown > renders correctly when completed 1`] = `
.c1 {
  display: flex;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  transform: rotate(0deg);
}

.c4 svg {
  color: #9d927b!important;
}

.c8 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  transform: rotate(0deg);
}

.c8 svg {
  color: #317a4c!important;
}

.c9 {
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
  text-align: left;
  color: #317a4c;
}

.c5 {
  width: 0;
  overflow: hidden;
  transition: width 0.6s ease-in;
  width: 100%;
}

.c7 {
  display: flex;
  align-items: center;
  gap: 4px;
}

.c0 {
  display: flex;
  flex-direction: column;
  position: relative;
  width: 100%;
  border: 0;
}

.c2 {
  border: none;
  border-radius: 0;
  font-size: 16px;
  appearance: none;
  outline: none;
  width: 100%;
  height: 32px;
  color: #636768;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 12px;
  padding: 18px 14px;
  border: 2px solid #dad2c4;
  height: auto;
}

.c2:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c2 :not(:disabled):hover,
.c2 :not(:disabled):focus,
.c2 :not(:disabled):focus-visible,
.c2 :not(:disabled):checked {
  border-color: #9d927b;
}

.c3 {
  position: absolute;
  z-index: 1;
  right: 15px;
  pointer-events: none;
  color: #9d927b;
}

@media (min-width: 0px) {
  .c1 {
    align-items: center;
  }
}

@media (min-width: 0px) {
  .c6 {
    margin-top: 8px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class=""
    >
      <div
        class="c1"
        style="position: relative;"
      >
        <select
          class="c2"
          id="MM_SMORES_1"
        >
          <option
            disabled=""
            hidden=""
            value=""
          />
          <option
            value="option1"
          >
            Option 1
          </option>
          <option
            value="option2"
          >
            Option 2
          </option>
        </select>
        <div
          class="c3"
        >
          <span
            class="c4"
            data-testid="caret-container"
          >
            <svg
              fill="none"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.66 8.248a1 1 0 00-1.412.095L12 15.482l-6.248-7.14a1 1 0 10-1.504 1.317l7 8a.995.995 0 001.504 0l7-8a1 1 0 00-.093-1.411z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6 c7"
      >
        <span
          class="c8"
          data-testid="included-container"
        >
          <svg
            fill="none"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
            <path
              d="M18.4808 6.924C18.1605 6.6585 17.6895 6.70275 17.424 7.02L10.3552 15.5025L6.41625 12.8768C6.0705 12.6458 5.6055 12.7395 5.376 13.0845C5.14575 13.4288 5.23875 13.8953 5.58375 14.1248L10.0837 17.1248C10.2112 17.2088 10.356 17.25 10.4992 17.25C10.7152 17.25 10.9282 17.157 11.0767 16.98L18.5768 7.98C18.8415 7.662 18.798 7.18875 18.4808 6.924V6.924Z"
              fill="white"
            />
          </svg>
        </span>
        <p
          class="c9"
          cursor="inherit"
          title=""
        >
          Complete
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown > renders correctly with a list of items 1`] = `
.c1 {
  display: flex;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  transform: rotate(0deg);
}

.c4 svg {
  color: #9d927b!important;
}

.c0 {
  display: flex;
  flex-direction: column;
  position: relative;
  width: 100%;
  border: 0;
}

.c2 {
  border: none;
  border-radius: 0;
  font-size: 16px;
  appearance: none;
  outline: none;
  width: 100%;
  height: 32px;
  color: #636768;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 12px;
  padding: 18px 14px;
  border: 2px solid #dad2c4;
  height: auto;
}

.c2:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c2 :not(:disabled):hover,
.c2 :not(:disabled):focus,
.c2 :not(:disabled):focus-visible,
.c2 :not(:disabled):checked {
  border-color: #9d927b;
}

.c3 {
  position: absolute;
  z-index: 1;
  right: 15px;
  pointer-events: none;
  color: #9d927b;
}

@media (min-width: 0px) {
  .c1 {
    align-items: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class=""
    >
      <div
        class="c1"
        style="position: relative;"
      >
        <select
          class="c2"
          id="MM_SMORES_0"
        >
          <option
            disabled=""
            hidden=""
            value=""
          />
          <option
            value="option1"
          >
            Option 1
          </option>
          <option
            value="option2"
          >
            Option 2
          </option>
        </select>
        <div
          class="c3"
        >
          <span
            class="c4"
            data-testid="caret-container"
          >
            <svg
              fill="none"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.66 8.248a1 1 0 00-1.412.095L12 15.482l-6.248-7.14a1 1 0 10-1.504 1.317l7 8a.995.995 0 001.504 0l7-8a1 1 0 00-.093-1.411z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown > renders correctly with an error message 1`] = `
.c1 {
  display: flex;
}

.c5 {
  display: flex;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  transform: rotate(0deg);
}

.c4 svg {
  color: #9d927b!important;
}

.c6 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  transform: rotate(0deg);
}

.c6 svg {
  color: #d03c30!important;
}

.c7 {
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
  text-align: left;
  color: #d03c30;
}

.c0 {
  display: flex;
  flex-direction: column;
  position: relative;
  width: 100%;
  border: 0;
}

.c2 {
  border: none;
  border-radius: 0;
  font-size: 16px;
  appearance: none;
  outline: none;
  width: 100%;
  height: 32px;
  color: #636768;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 12px;
  padding: 18px 14px;
  border: 2px solid #d03c30;
  height: auto;
}

.c2:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c3 {
  position: absolute;
  z-index: 1;
  right: 15px;
  pointer-events: none;
  color: #9d927b;
}

@media (min-width: 0px) {
  .c1 {
    align-items: center;
  }
}

@media (min-width: 0px) {
  .c5 {
    margin-top: 8px;
  }
}

@media (min-width: 0px) {
  .c5 {
    align-items: center;
  }
}

@media (min-width: 0px) {
  .c5 {
    gap: 4px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class=""
    >
      <div
        class="c1"
        style="position: relative;"
      >
        <select
          class="c2"
          id="MM_SMORES_2"
        >
          <option
            disabled=""
            hidden=""
            value=""
          />
          <option
            value="option1"
          >
            Option 1
          </option>
          <option
            value="option2"
          >
            Option 2
          </option>
        </select>
        <div
          class="c3"
        >
          <span
            class="c4"
            data-testid="caret-container"
          >
            <svg
              fill="none"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.66 8.248a1 1 0 00-1.412.095L12 15.482l-6.248-7.14a1 1 0 10-1.504 1.317l7 8a.995.995 0 001.504 0l7-8a1 1 0 00-.093-1.411z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="c5"
    >
      <span
        class="c6"
        data-testid="warning-container"
      >
        <svg
          fill="none"
          height="100%"
          viewBox="0 0 25 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.001 6a1 1 0 00-1 1v6a1 1 0 102 0V7a1 1 0 00-1-1zm0 10.005a1.003 1.003 0 000 2.005 1 1 0 001-1V17a.995.995 0 00-1-.995zm11.42-.503l-7.64-13.316A4.31 4.31 0 0012.001 0a4.31 4.31 0 00-3.78 2.186L.581 15.502a4.288 4.288 0 00.01 4.327A4.313 4.313 0 004.36 22h15.28a4.313 4.313 0 003.771-2.171 4.288 4.288 0 00.01-4.327zm-1.739 3.324A2.333 2.333 0 0119.642 20H4.361a2.331 2.331 0 01-2.042-1.174 2.306 2.306 0 01-.005-2.328l7.64-13.316A2.332 2.332 0 0112.001 2c.856 0 1.621.441 2.046 1.182l7.64 13.316c.418.73.417 1.6-.005 2.328z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="c7"
        cursor="inherit"
        title=""
      >
        This is an error
      </span>
    </div>
  </div>
</div>
`;
